core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -0,0 +1,346 @@
+package edu.asu.c3simulator.widgets;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.badlogic.gdx.Game;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+/**
+ * This class creates a navigation panel by creating the buttons and linking them with
+ * their screens. It also supports having one level of sub-buttons to a parent button. A
+ * parent button will only show and hide sub-buttons when clicked. Only one of the parent
+ * buttons can show its sub-buttons at the same time. If a parent button is clicked, the
+ * already extended sub-buttons will collapse before extending the new set of sub-buttons.
+ * 
+ * @author Alyahya, Mohammed
+ */
+public class NavigationPanel extends Table
+{
+	private class NavigationButtonListener extends ClickListener
+	{
+		String screenName;
+		Screen targetedScreen;
+		
+		public NavigationButtonListener(String screenName, Screen targetedScreen)
+		{
+			this.screenName = screenName;
+			this.targetedScreen = targetedScreen;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			System.out.println(screenName);
+			game.setScreen(targetedScreen);
+		}
+	}
+	
+	private class CollapsibleButtonListener extends ClickListener
+	{
+		String screenName;
+		
+		public CollapsibleButtonListener(String screenName)
+		{
+			this.screenName = screenName;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			if (displayedCollapsibleButton.equalsIgnoreCase(screenName))
+				displayedCollapsibleButton = "";
+			else
+				displayedCollapsibleButton = screenName;
+			recreateNavigationPanel();
+		}
+	}
+	
+	private Game game;
+	private Skin skin;
+	private LinkedHashMap<String, Screen> buttons;
+	private LinkedHashMap<String, LinkedHashMap<String, Screen>> subButtons;
```

Comment:
1: I didn't mean that he will have access to them. I meant that I had to use it because of the way I implemented this class and using it was mostly for the purpose of letting its user have easiest control of the Navigation panel object possible.

2: Yeah. My idea was that (for example) the implementer of the Product screens could have the ability to only show the Current product button on the navigation panel after at least one product is launched and also show the retired product button after at least one product is retired. So, to enable this I had to make it so that adding or removing buttons will have real-time effect on the navigation panel.

2.1: It might be better. I didn't think of it before.

3: I don't understand how that is possible with button listeners. Remember that what I mean by user here is the user of the class, not the  user of the GUI, which is the coder who manipulate the object in his code to get the desired GUI they want.

3.1 & 3.2: yeah both of those still stand. I am using button listeners for those functions but adding a button and removing it from the navigation panel itself is not something that can be done by interacting with the GUI but through the code, that is why I said I couldn't use button listeners for them since there is nothing to listen to.

No, I mean the navigation panel's own code. Right now in order to allow real time manipulation of the panel, like I mentioned in 2, I am going through all the hash maps and recreating them again whenever a new button is added or is removed (I know it is not the best way to go about it but I couldn't come with a better idea).